File: app/src/main/java/com/tangbba/simplegithubforjava/api/model/GithubAccessToken.java
package com.tangbba.simplegithubforjava.api.model;

import com.google.gson.annotations.SerializedName;

public class  GithubAccessToken {

    @SerializedName("access_token")
    public final String mAccessToken;

    @SerializedName("scope")
    public final String scope;

    @SerializedName("token_type")
    public final String mTokenType;

    public GithubAccessToken(String accessToken, String scope, String tokenType) {
        mAccessToken = accessToken;
        this.scope = scope;
        mTokenType = tokenType;
    }

    public String getAccessToken() {
        return mAccessToken;
    }

    public String getScope() {
        return scope;
    }

    public String getTokenType() {
        return mTokenType;
    }
}

File: app/src/main/kotlin/com/tangbba/simplegithubforjava/api/model/GithubAccessToken.kt
package com.tangbba.simplegithubforjava.api.model

import com.google.gson.annotations.SerializedName

class GithubAccessToken(@field:SerializedName("access_token") val accessToken: String,
                        @field:SerializedName("scope") val scope: String,
                        @field:SerializedName("token_type") val tokenType: String)
